jib-core/src/main/java/com/google/cloud/tools/jib/cache/CacheStorageReader.java

```diff
@@ -110,7 +110,7 @@
 
       if (schemaVersion == 1) {
         return Optional.of(
-            new ManifestAndConfig(
+            new ManifestAndConfigTemplate(
```

Comment:
It seems strange that we would generate a JsonTemplate type from separately parsed out jsonTemplates? Is this something that we would need to change eventually? 

Can we go directly to ManifestAndConfigTemplate directly from file?

Reply:
In the next PR, all of these parsing and conversion operations will be gone, and this `retrieveMetadata()` will directly de-serialize and return `ImageMetadataTemplate` without any processing.